@types/ApiClient.d.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+type RequestOptions = {
+    headers?: Record<string, string>;
+    body?: Record<string, string>;
+}
+
+type Headers = Record<string, string>;
+
+type Response = {
+    headers?: Headers;
+    data?: Record<string, string>;
+}
```

Comment:
headers: ok -> not optional
data: ok? -> this could also be an array, plain text or a blob -> maybe consider adding some functions like base64(), json(), text() to conform with the fetch API
status: missing
redirects: missing -> this should be an array with all url's